packages/components/src/components/header/header.ts

```diff
@@ -0,0 +1,158 @@
+ */
+@customElement('sd-header')
+export default class SdHeader extends SolidElement {
+  @property({ reflect: true, type: Boolean }) fixed = false;
+  @property({ attribute: 'auto-spacing', reflect: true, type: Boolean }) autoSpacing = true;
+
+  private refHeaderElement?: HTMLElement;
+  private resizeObserver?: ResizeObserver;
+
+  connectedCallback(): void {
+    super.connectedCallback();
+    this.setBodySpacing = this.setBodySpacing.bind(this);
+    this.addResizeObserver = this.addResizeObserver.bind(this);
+  }
+
+  firstUpdated(): void {
+    this.setBodySpacing();
+    this.addResizeObserver();
+  }
+
+  disconnectedCallback(): void {
+    super.disconnectedCallback();
+    if (this.resizeObserver) {
+      this.resizeObserver.disconnect();
+      this.resizeObserver = undefined;
+    }
+  }
+
+  updated(changedProperties: PropertyValues): void {
+    super.updated(changedProperties);
+
+    if (changedProperties.has('autoSpacing')) {
+      this.setBodySpacing();
+      if (this.autoSpacing) {
+        this.addResizeObserver();
+      } else if (this.resizeObserver) {
+        this.resizeObserver.disconnect();
+        this.resizeObserver = undefined;
+      }
+    }
+  }
+
+  @debounce(100)
+  private onResize(): void {
+    if (this.autoSpacing) {
+      this.setBodySpacing();
+    }
+  }
+
+  private addResizeObserver(): void {
+    this.resizeObserver = new ResizeObserver(() => {
+      this.onResize();
+    });
+
+    if (this.refHeaderElement) {
+      this.resizeObserver.observe(this.refHeaderElement);
+    }
+  }
+
+  private setBodySpacing(): void {
+    if (this.autoSpacing && this.refHeaderElement) {
+      document.body.style.paddingTop = `${this.refHeaderElement.clientHeight}px`;
+    } else {
+      document.body.style.paddingTop = '';
+    }
+  }
+
+  render() {
+    return html`
+      <header
+        class=${cx('w-screen bg-white relative', this.fixed && 'fixed-shadow w-screen')}
+        role="banner"
+        @slotchange=${this.handleSlotChange}
+      >
+        <div part="main" class="relative">
+          <slot></slot>
+        </div>
+      </header>
+    `;
+  }
+
+  private handleSlotChange(event: Event): void {
+    const slot = event.target as HTMLSlotElement;
+    if (slot.assignedElements().length > 0) {
+      this.refHeaderElement = event.currentTarget as HTMLElement;
+      this.setBodySpacing();
+      this.addResizeObserver();
+    }
+  }
+
+  static styles = [
+    SolidElement.styles,
+    componentStyles,
+    css`
+      :host {
+        display: block;
+        z-index: 65536;
+        position: absolute;
+        top: 0;
+        left: 0;
+      }
+
+      :host([fixed]) {
+        position: fixed;
+      }
+
+      .fixed-shadow {
+        box-shadow: 0 4px 2px -2px gray;
+      }
+
+      [part='main'] {
+        margin: 0 auto;
+        width: var(--sd-header-inner-width, calc(100vw - 2 * var(--sd-header-padding-x, 16px)));
```

Comment:
Aha, now I see, this is what makes the width overflow in the stories.  It looks strange in storybook but makes sense for the component.  Maybe we should just add a comment in the story to make it obvious for developers